### PR TITLE
fix(apk): preserve Astro-built apk docs page across overlay assembly

### PR DESCRIPTION
**Pull request by omercnet**
## Summary

`https://verity.supply/apk/` returns 404 in production. Root cause:
`.github/scripts/assemble-apk-repository.sh` runs `rm -rf "\$OUTPUT_DIR"`
(default `site/dist/apk`) **after** the Astro build, wiping out the
`index.html` + `index.md` emitted by `site/src/pages/apk/index.astro`.

Effect: every visitor that clicks the homepage's prominent **APK Repository
Preview → View docs** CTA lands on GitHub Pages' generic 404. The
experimental-caveats page that was supposed to explain what's published, the
key fingerprint, and that the repository may be empty is silently deleted on
every nightly run. The deleted page even contains the line *"The repository
may be empty or return 404 until publish and verification complete"* — which
is now unreachable.

## Fix

Replace the blanket `rm -rf` with targeted cleanup of only the artifacts this
script manages:

- `.no-apks-found` marker
- `*.rsa.pub` keys it publishes
- Arch directories listed in the new `SUPPORTED_ARCHES` array (kept in sync
  with the existing `detect_arch` allowlist)

Anything else in the output directory — notably the Astro-built docs page —
is preserved.

## Tests

Three new tests in `scripts/apk_repository_test.go`:

- `TestAssembleAPKRepositoryPreservesPreexistingDocsWhenNoPackages`
- `TestAssembleAPKRepositoryPreservesDocsWhenPackagesPresent`
- `TestAssembleAPKRepositoryCleansStaleArchAndMarker`

All existing tests continue to pass.

## Related

- Companion PRs (independent, opt-in mergeable in any order):
  - Site nav / layout / missing index pages
  - Site WCAG contrast + readability
  - Responsive mobile tables for compliance and VulnTable

## Test plan

```bash
go test ./scripts/ -run TestAssembleAPK -v
```

---
## Summary by cubic
Preserves the Astro-built APK docs page during overlay assembly so https://verity.supply/apk/ no longer returns 404. Tightens cleanup to script-owned artifacts, unifies arch allowlist, and hardens cleanup paths.

- **Bug Fixes**
  - Replace `rm -rf $OUTPUT_DIR` with targeted cleanup of `.no-apks-found`, `$KEY_NAME.pub`, and arch dirs from `SUPPORTED_ARCHES`; preserve docs in `site/dist/apk` (e.g., `index.html`, `index.md`).
  - Make `detect_arch` iterate `SUPPORTED_ARCHES` to prevent allowlist drift; scope key cleanup to `$KEY_NAME.pub` and protect unrelated `.rsa.pub` files.
  - Brace cleanup paths using `${VAR:?}/...` for consistency and safety.
  - Add tests for docs preservation, scoped key cleanup, and arch allowlist (including a negative case); update the `detect_arch` test to assert failures with `require.Error` instead of ignoring the error.

<sup>Written for commit b89a222c5cf872028900aa9b1897222beba679f7. Summary will update on new commits. <a href="https://cubic.dev/pr/verity-org/verity/pull/485?utm_source=github">Review in cubic</a></sup>

### CODE REVIEW

File: .github/scripts/assemble-apk-repository.sh [hunk 1]
```diff
@@ -15,9 +15,19 @@ Defaults:
...
+(e.g. index.html, index.md from a prior Astro build) are left untouched.
+Only managed artifacts (.no-apks-found, arch directories, the public key,
+and APKINDEX.tar.gz files) are removed at the start of each run.
 USAGE
 }
 
+# Architectures the script writes to OUTPUT_DIR. Kept in sync with detect_arch
+# below — any arch listed here may have a top-level directory removed on a
+# fresh run.
+SUPPORTED_ARCHES=(x86_64 aarch64 armv7 armhf ppc64le s390x riscv64)
```
[the remark this answers is not in the document]
Comment (omercnet) [reply .github/scripts/assemble-apk-repository.sh:30]: Fixed in c9f9a5418d. `detect_arch` now iterates `SUPPORTED_ARCHES` instead of its own hardcoded `case`, so a new arch only needs to be added in one place. Regression: `TestAssembleAPKRepositoryDetectArchUsesSupportedArchesAllowlist` is parameterised over every arch in the allowlist plus a negative case for an unsupported arch.

File: .github/scripts/assemble-apk-repository.sh [hunk 2]
```diff
@@ -110,8 +120,15 @@ mapfile -t APKS < <(
   done | sort -u
 )
 
-rm -rf "$OUTPUT_DIR"
+# Clean only artifacts this script manages. Preserve any other files (notably
+# index.html and index.md emitted by the Astro build) so the published page
+# still explains the repository even when no .apk files were produced.
 mkdir -p "$OUTPUT_DIR"
+rm -f "$OUTPUT_DIR/.no-apks-found"
+rm -f "$OUTPUT_DIR"/*.rsa.pub
```
[the remark this answers is not in the document]
Comment (omercnet) [reply .github/scripts/assemble-apk-repository.sh]: Fixed in c9f9a5418d. Cleanup now scopes to `"$KEY_NAME.pub"` instead of the `*.rsa.pub` glob. Regression: `TestAssembleAPKRepositoryDoesNotRemoveUnrelatedRSAPubFiles` drops a sibling `.rsa.pub` next to the managed key and verifies it survives while the managed key is still cleaned.

File: .github/scripts/assemble-apk-repository.sh [hunk 3]
```diff
@@ -110,8 +121,15 @@ mapfile -t APKS < <(
   done | sort -u
 )
 
-rm -rf "$OUTPUT_DIR"
+# Clean only artifacts this script manages. Preserve any other files (notably
+# index.html and index.md emitted by the Astro build) so the published page
+# still explains the repository even when no .apk files were produced.
 mkdir -p "$OUTPUT_DIR"
+rm -f "$OUTPUT_DIR/.no-apks-found"
+rm -f "$OUTPUT_DIR/$KEY_NAME.pub"
```
[the remark this answers is not in the document]
Comment (omercnet) [reply .github/scripts/assemble-apk-repository.sh]: Fixed in b89a222c5c. All three cleanup paths now use the same `${OUTPUT_DIR:?}/...` form with braced variables (`${KEY_NAME}.pub`, `${arch}`). The `:?` guard is consistent across the three lines.

File: scripts/apk_repository_test.go [hunk 4]
```diff
@@ -70,6 +70,141 @@ func TestAssembleAPKRepositoryNoPackagesCreatesMarker(t *testing.T) {
...
+	// apk index will fail on the fake .apk; we only care the script reached the
+	// preservation point and did not delete the existing docs page before
+	// failing.
+	_, err := runGithubScript(t, repoRoot, "assemble-apk-repository.sh", "--output", outputDir, "apk-artifacts")
+	if err != nil {
+		// Failures from `apk index` against a fake archive are expected in the
+		// test environment. The preservation guarantee is independent of that
+		// outcome, so we ignore the error and assert on the file state below.
+		_ = err
+	}
```
[the remark this answers is not in the document]
Comment (omercnet) [reply scripts/apk_repository_test.go:109]: Fixed in b89a222c5c. The fake `.apk` does cause the script to fail at `require_tool apk` or `apk index`, so the discard is now an explicit `require.Error(t, err, "fake .apk should cause apk index (or require_tool) to fail")`. The test now reads as `"fails for a non-detect_arch reason"` rather than silently dropping the error.
